ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -896,10 +896,11 @@ private void cleanupExpiredInFlightRequests(RequestRegistrationCallback<GetOpera
                   chunkBlobId));
           requestRegistrationCallback.registerRequestToDrop(correlationId);
           inFlightRequestsIterator.remove();
-        } else {
-          // the entries are ordered by correlation id and time. Break on the first request that has not timed out.
-          break;
         }
+        // Note: Even though the entries are ordered by correlation id and their creation time, we cannot break out of
+        // the while loop when we find an unexpired entry. This is because time outs for all requests may not be equal
+        // now. For example, we assign higher time outs when we find that there are lot of outstanding requests by
+        // letting Network client dynamically update the timeout via RequestInfo#incrementNetworkTimeOutMs.
```

Comment:
This comment is not clear. Is the time out per operation or per request within the operation ? If we find the first request that has not yet timed, can't we assert that subsequent ones would not have timed out since their expiry would be further in the future ?

Reply:
Yeah, in almost all of the cases, when we find a request that has not yet timed out, subsequent requests would also be valid. But, since the time outs of all requests is not fixed now, it is possible that later request in the queue has shorter time out. This helps to clean up such requests. I think performance wise it might be okay since the map would only hold few elements at a time (i.e. number of outstanding requests which should be like 3 or 4). 

Updated comments in code to make them more clear.